sdk/azcore/arm/runtime/pipeline.go

```diff
@@ -34,7 +35,7 @@ func NewPipeline(module, version string, cred azcore.TokenCredential, plOpts azr
 	})
 	perRetry := make([]azpolicy.Policy, len(plOpts.PerRetry), len(plOpts.PerRetry)+1)
 	copy(perRetry, plOpts.PerRetry)
-	plOpts.PerRetry = append(perRetry, authPolicy)
+	plOpts.PerRetry = append(perRetry, authPolicy, exported.PolicyFunc(httpTraceNamespacePolicy))
```

Comment:
future-proofness:

what's the order of auth, tracing and logging policies? 

We should put logging  under tracing, i.e.

auth -> .... -> start HTTP span -> .... -> log HTTP request
                      end HTTP span <- .... <- log HTTP response

so that we trace 401 attempts and can in future start stamping trace context on logs to correlate between them.

Reply:
fixed.